ros/api/v0/hosts.py

```diff
@@ -103,14 +105,19 @@ def get(self):
             'instance_type', 'cloud_provider',
             'rule_hit_details', 'state', 'number_of_recommendations']
         for row in query_results:
-            system_dict = row.System.__dict__
-            host = {skey: system_dict[skey] for skey in system_columns}
-            host['account'] = row.RhAccount.account
-            host['display_performance_score'] = row.PerformanceProfile.display_performance_score
-            host['idling_time'] = row.PerformanceProfile.idling_time
-            host['io_wait'] = row.PerformanceProfile.io_wait
-            hosts.append(host)
-
+            try:
+                system_dict = row.System.__dict__
+                host = {skey: system_dict[skey] for skey in system_columns}
+                host['account'] = row.RhAccount.account
+                host['display_performance_score'] = row.PerformanceProfile.display_performance_score
+                host['idling_time'] = row.PerformanceProfile.idling_time
+                host['io_wait'] = row.PerformanceProfile.io_wait
+            except Exception as err:
+                LOG.error(
+                    'An error occured while fetching the host. %s',
+                    repr(err)
+                )
+                hosts.append(host)
```

Comment:
This should be in the Try block.

Reply:
Yep, right! Made a typo in the order and didn't test it so couldn't verify.